### ISSUE TITLE
Command to disable beeping

### ISSUE THREAD

**Issue by Gabeki17** [tomikaa87/gree-remote#44]
It would be very useful, as I am controling my airconditioner automatically.
Quit anoying.

Anyone have tried to do this already?

Regards,
Gábor

**Comment by tomikaa87** [comment]
This would be an interesting feature, but unfortunately I haven't seen anything related in the disassembled mobile app.

**Comment by tomikaa87** [comment]
I'm just brainstorming, but maybe it's worth a shot to disassemble a firmware file to find out the list of available commands. However I think the beeping "feature" is a part of the AC's main logic board and not the WiFi controller's, so it's possible that this is a a hardware limitation and only de-soldering the beeper would help.

**Comment by Gabeki17** [comment]
> I'm just brainstorming, but maybe it's worth a shot to disassemble a firmware file to find out the list of available commands. However I think the beeping "feature" is a part of the AC's main logic board and not the WiFi controller's, so it's possible that this is a hardware limitation and only de-soldering the beeper would help. (In that case, **you must be cautious, there is ⚠️⚡︎ high voltage inside ⚠️⚡︎** and you can easily hurt yourself or damage the unit.)

I think you are right!

On all.

Now I am optimising my code so it is not commanding the AC too much. 
By this, the beepeing went already to a tolerable level.

But in the end I will desoldier the beeper I think.

Also on sending the temperature in. I have foud it possible. it is taking for like a second only. right after refreshing and is gone. ... Oh well

**Comment by DaniilMir** [comment]
You may also put a piece of tape on beeper itself or cover it with a bubble gum or something else. it will reduce the beep level significantly and will not void the warranty)
But yes, it would be a nice feature to have in application. 
Maybe we should post it on app store page? I tried to contact their devs for other questions but had no success.

**Comment by mbarrerar** [comment]
It would be a great feature, currently it is impossible to disable the beep

**Comment by mlesniew** [comment]
Hi all,

great to see that I'm not the only one who finds the beeps annoying.  I've investigated this myself for a long time too, so let me share my findings.

There doesn't seem to be a command to directly disable the beeping via software.  However, there is a way to change at least some parameters (e.g. target temperature, wind speed) without the AC emitting a beep using programming/schedule mode in the app as @mbarrerar suggests.

Here's how you do it manually:
* In the app, open the AC you want to control
* Select the last icon on the bottom with the clock icon.
* Add a new schedule (e.g. to turn on the AC).
* Set the desired parameters (e.g. target temperature)
* Set the time to one minute ahead of the current time. 
* Make sure to select (single-time schedule) -- not sure how this setting is labeled in the English version of the app, but it will make sure this will not be a recurring, periodic schedule.   
* Save

After one minute the AC will enter the programmed mode without beeping.

Now doing some research on the protocol I found that:
* The protocol actually allows setting the programmed time with second precision.  This means that if the message would be sent programmatically, it should be possible to program the AC to switch the mode after a few seconds, not after a minute)
* It should be possible to set extra parameters this way too.  The app only allows programming a small set of parameters this way, but internally the protocol allows specifying any parameter.

Now I haven't written any code to empirically prove the above points, I would have to experiment a bit, but it looks promising.  

I see one potential problem with this though.  The programmed schedules are probably stored in the device's flash memory.  Using the described trick would mean the programming feature is used a lot for single-time schedules, which could cause a lot of flash wear.     

Regarding physical ways to silence the buzzer, I tried what @tomikaa87 and @DaniilMir suggested.  

The bad news is that putting tape over the buzzer on the main circuit didn't help at all.  In many similar buzzers I've seen in electronics this trick does work, but not in this one.  It's caused by the way the buzzer is built.

The good news is that it's relatively easy to remove the membrane of the buzzer.  The buzzer has two plastic parts -- the base (which is soldered to the circuit board) and a cap with a hole that is put on top.  Between this plastic parts is the metal membrane.  The plastic cap can be easily removed, I used a flat head screwdriver to lift it and it popped out, without breaking the plastic.  I could then remove the membrane, it was not attached to the plastic. 

The best part about this is that this is completely reversible.  You can easily put the membrane back into the buzzer and it will start beeping again.  It also shouldn't void the warranty, because it doesn't leave any traces.  

Still, I didn't like my AC without any beeps.  I realized that while the beeps are annoying when the AC is controlled from a script, they are actually useful when the remote is used.  When using the remote, the beep is a confirmation that the signal was received properly (which is not always the case, because it's IR).

**Comment by mlesniew** [comment]
Sorry guys, I was wrong.  There is a parameter that turns off the beep programmatically.  Just after I posted the last comment I found another way to explore supported parameters and found it. 
 
The parameter that controls the beep is `Buzzer_ON_OFF`.  Set it to `1` to disable the beep.  

For example, to set the target temperature to 21 degrees using the Python script run:

```bash
python3 gree.py -c <ip> -i <id> -k <key> set SetTem=21 Buzzer_ON_OFF=1
```

The new target temperature setting will be applied, but the buzzer will not beep.  

`Buzzer_ON_OFF` is not a persistent setting, the AC doesn't remember it.  Subsequent `set` commands will still cause a beep unless they're sent with `Buzzer_ON_OFF=1` too. 

I tested this with my Gree Pular unit.

**Comment by raythefourth** [comment]
I'm interested in silencing the beeping as well. I want to turn off the Panel LED when I go to sleep (with an automation), but I don't want to wake anybody else up with the loud beeping. 

I was able to send some commands without the beep for example:
python3 gree.py -c [ip] -i [mac] -k [key] -e GCM set Lig=0 Buzzer_ON_OFF=1
python3 gree.py -c [ip] -i [mac] -k [key] -e GCM set Lig=1 Buzzer_ON_OFF=1

However, for me it would only silence every-other beep. The first command would be executed silently, and the second command would always beep. I tried a few different things but can't figure out why it only works half the time.

Any ideas on this? Thanks!

**Comment by mlesniew** [comment]
Hey @raythefourth I tried the commands you wrote in your comment and my AC is quiet every time.

What AC model have you tried this with?

Perhaps in your model sending `Buzzer_ON_OFF` toggles the beeping setting on and off... 

Try reading the `Buzzer_ON_OFF` value after each `set` command, using: `python3 gree.py ... get Buzzer_ON_OFF` and see if its value changes after you attempt to change it.

In my case `Buzzer_ON_OFF` is always set to 0, no matter what the last set command was.

**Comment by ov1d1u** [comment]
I'm seeing the same behavior like @raythefourth. `get Buzzer_ON_OFF` after a `set Buzzer_ON_OFF=1` shows inconsistent results, sometimes it shows `Buzzer_ON_OFF = 1` and then `Buzzer_ON_OFF = 0`, but most of the times it only shows `Buzzer_ON_OFF = 0`.

My AC model is Bora A4 Silver GWH12AABXB-K6DNA4A.

[gree_commands.txt](https://github.com/user-attachments/files/17774856/gree_commands.txt)

**Comment by mlesniewski-camlin** [comment]
@ov1d1u For me the `Buzzer_ON_OFF` value switches back to `0` too, it only stays at `1` for a few seconds at most.  But that doesn't matter -- what matters is that if I sent a command with `Buzzer_ON_OFF=1` the beep is always disabled.

**Comment by mlesniew** [comment]
@ov1d1u For me the Buzzer_ON_OFF value switches back to 0 too, it only stays at 1 for a few seconds at most. But that doesn't matter -- what matters is that if I sent a command with Buzzer_ON_OFF=1 the beep is always disabled.

**Comment by ov1d1u** [comment]
For me, it is silenced only every other command. So if I run:

`python3 gree.py -c 192.168.0.174 -i <mac> -k <key> -e GCM set Lig=0 Buzzer_ON_OFF=1`

it does not beep, but if I run again the same command:

`python3 gree.py -c 192.168.0.174 -i <mac> -k <key> -e GCM set Lig=0 Buzzer_ON_OFF=1`

it beeps. Next time it won't beep, then it will beep again, and so on.

What is the version of the WiFi module firmware (you can see it in the GREE+ app)? Mine is V1.23, maybe it is related to that.

**Comment by mlesniew** [comment]
The firmware my unit is running is at 2.06, but I believe every model has an independent firmware.

**Comment by boris-bc** [comment]
> For me, it is silenced only every other command. So if I run:
> 
> `python3 gree.py -c 192.168.0.174 -i <mac> -k <key> -e GCM set Lig=0 Buzzer_ON_OFF=1`
> 
> it does not beep, but if I run again the same command:
> 
> `python3 gree.py -c 192.168.0.174 -i <mac> -k <key> -e GCM set Lig=0 Buzzer_ON_OFF=1`
> 
> it beeps. Next time it won't beep, then it will beep again, and so on.
> 
> What is the version of the WiFi module firmware (you can see it in the GREE+ app)? Mine is V1.23, maybe it is related to that.

I have the same firmware version 1.23 and I have the same behavior. My indoor units are Fairy II and this beeping sound is super annoying at night. Does anyone have any email to Gree developers to tell them about this bug?

**Comment by namezys** [comment]
Today I spend couple hours to check different command sequences in attention to get solution. Unsuccessful, every second command had a beep.

However, there was a firmware update from `1.23` to `1.31`. After update it works.
In-door unit: `Comfort X`

And on other unit `Comfort Pulse` everything works as well without firmware updates.

`Buzzer_ON_OFF=1` works again

**Comment by ov1d1u** [comment]
I updated the WiFi module to 1.31 and I can confirm that `Buzzer_ON_OFF=1` works as expected now. Just in time for the summer.

**Comment by nvv-git** [comment]
Hello everyone,

I am using **Home Assistant** with **Node-RED** to automate my **Gree Lomo Inverter** (Model: **GWH09QC-K6DNB4F/I**). The Wi-Fi module firmware version is 1.12. Recently, I discovered an interesting behavior with the beeper.

A few days ago, I created a schedule in the official **Gree** app (Android) to turn off the air conditioner at a specific time. After doing this, I noticed that the beeper stopped sounding when commands were sent from both **Home Assistant** and the **Gree** app. However, when I use the physical remote, the beeper still sounds as usual.

This behavior persists for all commands sent from automation and the app, remaining completely silent. However, if I disconnect the power (unplug the unit), the beeper returns to normal for all commands, including those sent from **Node-RED** and the app.

It seems that the **silent mode** is maintained until the AC loses power.
I also tried sending the **Buzzer_ON_OFF=1** parameter via **Node-RED**, but it didn't work. Only the schedule from the app seems to trigger this silent mode.

Has anyone else experienced this? Is there a way to replicate this programmatically through **Node-RED** or **MQTT**? Or perhaps a command that mimics the effect of the scheduled action? I would appreciate any ideas or suggestions.

**Thanks in advance!**

**Comment by namezys-at-getpatch** [comment]
There is no direct connection betwean `Buzzer_ON_OFF` and beep.
As I understand, beep is a reaction on any external command like IR command or WiFi command.
I don't know where a beeper is located but I suppose that beeper is controlled by main logic instead of WiFi module. Also, I suppose that WiFi module is separated from main logic.

My suggestion that each command is just a set of parameters and one of this parameter is suppression of sound.
In some versions of WiFi firmware this suppression is sent as should not. The source of command is does not matter, it can be HA, Node-Red, MQQT etc. The main issue is buggy firmware.

However, it's a good to know that have schedule can affect it. 

As a solution (except fixed firmware) I can suggest to set 2 schedules as soon as device discovered. 

PS: actually different WiFi modules can have different firmware versions and different bugs

**Comment by zhouska** [comment]
> Today I spend couple hours to check different command sequences in attention to get solution. Unsuccessful, every second command had a beep.
> 
> However, there was a firmware update from `1.23` to `1.31`. After update it works. In-door unit: `Comfort X`
> 
> And on other unit `Comfort Pulse` everything works as well without firmware updates.
> 
> `Buzzer_ON_OFF=1` works again

Did you use the Gree or Ewpe app to upgrade? Mine doesn't show anything, so I had to MITM proxy it and subvert it to look at other gree servers for 1.31 update.

**Comment by zhouska** [comment]
> I updated the WiFi module to 1.31 and I can confirm that `Buzzer_ON_OFF=1` works as expected now. Just in time for the summer.

Did you use the Gree or Ewpe app to upgrade? Mine doesn't show anything, so I had to MITM proxy it and subvert it to look at other gree servers for 1.31 update.

**Comment by namezys** [comment]
> Did you use the Gree or Ewpe app to upgrade? Mine doesn't show anything, so I had to MITM proxy it and subvert it to look at other gree servers for 1.31 update.

Oficial Gree app.

I suppose it depends on the WiFi board.

**Comment by 23doors** [comment]
Seems to be heavily inconsistent. Got 3 AC units. 
All updated to newest available versions. 
One has 3.76
Another 2.06
And yet another one 3.75

Buzzer_ON_OFF=1 has zero effect on 2 of them. But when it's sent along with every command to the unit that sits on 2.06, it disables beep. 

Other 2 work just ignore it completely.

**Comment by namezys-at-getpatch** [comment]
This is not like setting. It must be send every command. 

Looks like they fixed it in the last firmware that was release last month. Probably, they haven't released for all boards.

* `v2.06`: I got update after installation, about 4-5 weeks ago (unit Comfort Pulse)
* `v1.31`: I got update about 2 week ago (unit Comfort X)

I suppose you will get update for others soon.

**Comment by zhouska** [comment]
> This is not like setting. It must be send every command.
> 
> Looks like they fixed it in the last firmware that was release last month. Probably, they haven't released for all boards.
> 
>     * `v2.06`: I got update after installation, about 4-5 weeks ago (unit Comfort Pulse)
> 
>     * `v1.31`: I got update about 2 week ago (unit Comfort X)
> 
> 
> I suppose you will get update for others soon.

yes, spot on, the silent buzzer command needs to be sent with every command  to the unit. I plan to update to 1.31 from 1.21 on one unit only and test - but I have to sideload the firmware download URL (already figured it out - my board doesn't have any officially published version on EU servers) with help of a mitm proxy - the board ID matches as far as I can see - so let's hope I won't brick anything...

**Comment by mahi2003** [comment]
I just started using `gree.py` to control my Heiwa Essentiel ZEN+ (HMIS2-25P-V2) air conditioner. I was excited to find this topic, because the beeping is very annoying. Unfortunately my unit comes with firmware 1.23 and shows the same bug as mentioned above: Silencing the AC works only every other command.

Member ov1d1u was able to update his 1.23 to 1.31 three weeks ago and it fixed the issue with `Buzzer_ON_OFF`.

I checked the app, but there's no firmware update available for my unit.

I was intrigued by zhouska's comment to attempt to make the unit load a firmware from a different region. I tried this, but did not succeed. I can see the AC successfully downloading the 1.31 firmware, but after that nothing happens. The update progress in the app times out and everything returns to normal. The unit is not bricked, but it is not updated either.

---

For completeness, here's what I tried:

* With `gree.py search --verbose` my unit returns `362001065279+U-WB05RT13V1.23.bin`.
* I checked https://eugrih.gree.com/wifiModule/Lastversion/362001065279, but no firmware is available:
```
{
  "r":405,
  "msg":"Firmware not found."
}
```
* However, https://kfgrih.gree.com/wifiModule/Lastversion/362001065279 returns 1.31!
```
{
  "CreateDate":"2025-02-25 07:44:09",
  "commProtVer":"V3.4.M",
  "desc":"ota upgrade",
  "forcedUpgrade":0,
  "r":200,
  "url":"http://kf.grih.gree.com/wifiModule/image/14376/807236",
  "ver":"1.31"
}
```
* Getting URL http://kf.grih.gree.com/wifiModule/image/14376/807236 downloads the firmware with following headers:
```
HTTP/1.1 200 OK
Date: Sat, 24 May 2025 17:10:53 GMT
Content-Type: application/octet-stream
Content-Length: 807236
Connection: keep-alive
Content-Disposition: attachment; filename="U-WB05RT13V1.31-17463628558052314052.bin"
Content-Range: bytes 0-807235/807236
FileMd5: 3f960fdfeaad1e344c14456a8caa2d59
```
* The filename contains the very same `U-WB05RT13` as reported by my AC. So, I was starting to get confident the firmware would be compatible.
* In mitmproxy I set `map_remote` = `|eugrih.gree.com/wifiModule|kfgrih.gree.com/wifiModule` to redirect the firmware update check from Europe to Asia (? not sure which region 'kf' is).
* With the phone's traffic going through mitmproxy, I checked for firmware updates. Success! This time firmware 1.31 is shown as available for my device.
* I started a packet capture of the air conditioner's IP address via ntopng on my pfSense router, just so that I could see what it does.
* I started the firmware update in the app.
* After a few minutes the progress got stuck at 99%. Another few minutes later the app returned to the previous screen as if nothing had happened.
* After inspecting the packet capture, I could see that the AC downloaded the firmware successfully from http://kf.grih.gree.com/wifiModule/image/14376/807236 in about 90 seconds, but nothing happened after that. The unit did not brick, nor did it update. It just works as before. So, there must be something with the downloaded file or headers that the AC did not like. Or I must be missing something else. I don't know what, though...

**Comment by zhouska** [comment]
> I just started using `gree.py` to control my Heiwa Essentiel ZEN+ (HMIS2-25P-V2) air conditioner. I was excited to find this topic, because the beeping is very annoying. Unfortunately my unit comes with firmware 1.23 and shows the same bug as mentioned above: Silencing the AC works only every other command.
> 
> Member ov1d1u was able to update his 1.23 to 1.31 three weeks ago and it fixed the issue with `Buzzer_ON_OFF`.
> 
> I checked the app, but there's no firmware update available for my unit.
> 
> I was intrigued by zhouska's comment to attempt to make the unit load a firmware from a different region. I tried this, but did not succeed. I can see the AC successfully downloading the 1.31 firmware, but after that nothing happens. The update progress in the app times out and everything returns to normal. The unit is not bricked, but it is not updated either.
> 
> For completeness, here's what I tried:
> 
> * With `gree.py search --verbose` my unit returns `362001065279+U-WB05RT13V1.23.bin`.
> * I checked https://eugrih.gree.com/wifiModule/Lastversion/362001065279, but no firmware is available:
> 
> ```
> {
>   "r":405,
>   "msg":"Firmware not found."
> }
> ```
> 
> * However, https://kfgrih.gree.com/wifiModule/Lastversion/362001065279 returns 1.31!
> 
> ```
> {
>   "CreateDate":"2025-02-25 07:44:09",
>   "commProtVer":"V3.4.M",
>   "desc":"ota upgrade",
>   "forcedUpgrade":0,
>   "r":200,
>   "url":"http://kf.grih.gree.com/wifiModule/image/14376/807236",
>   "ver":"1.31"
> }
> ```
> 
> * Getting URL http://kf.grih.gree.com/wifiModule/image/14376/807236 downloads the firmware with following headers:
> 
> ```
> HTTP/1.1 200 OK
> Date: Sat, 24 May 2025 17:10:53 GMT
> Content-Type: application/octet-stream
> Content-Length: 807236
> Connection: keep-alive
> Content-Disposition: attachment; filename="U-WB05RT13V1.31-17463628558052314052.bin"
> Content-Range: bytes 0-807235/807236
> FileMd5: 3f960fdfeaad1e344c14456a8caa2d59
> ```
> 
> * The filename contains the very same `U-WB05RT13` as reported by my AC. So, I was starting to get confident the firmware would be compatible.
> * In mitmproxy I set `map_remote` = `|eugrih.gree.com/wifiModule|kfgrih.gree.com/wifiModule` to redirect the firmware update check from Europe to Asia (? not sure which region 'kf' is).
> * With the phone's traffic going through mitmproxy, I checked for firmware updates. Success! This time firmware 1.31 is shown as available for my device.
> * I started a packet capture of the air conditioner's IP address via ntopng on my pfSense router, just so that I could see what it does.
> * I started the firmware update in the app.
> * After a few minutes the progress got stuck at 99%. Another few minutes later the app returned to the previous screen as if nothing had happened.
> * After inspecting the packet capture, I could see that the AC downloaded the firmware successfully from http://kf.grih.gree.com/wifiModule/image/14376/807236 in about 90 seconds, but nothing happened after that. The unit did not brick, nor did it update. It just works as before. So, there must be something with the downloaded file or headers that the AC did not like. Or I must be missing something else. I don't know what, though...

yes, this is more or less what I came across - slightly different approach but the firmware update hangs there at 99% for about two minutes or so, there is a 'firmware update failed' at the end (for a second or two - then it loads the same app screen). I'm not sure if it is because commProtVer or something else, but the firmware fails to flash apparently.

I will try with version 1.32 tomorrow, it is the version available on 'test' server, but it has same commProtVer as 1.31. There were reports of people being able to upgrade to 1.53 from 'test' server last year (for the same wifi board), this version has commProVer 3.0.0, both 1.21 and 1.23 have version set to 2.0.0 and 1.31 to 3.4M - but I'm not able to confirm if 1.53 silences the buzzer or not (yet)...

**Comment by zhouska** [comment]
> > Did you use the Gree or Ewpe app to upgrade? Mine doesn't show anything, so I had to MITM proxy it and subvert it to look at other gree servers for 1.31 update.
> 
> Oficial Gree app.
> 
> I suppose it depends on the WiFi board.

what is you wifi board?

**Comment by simonepsp** [comment]
I'm currently using a CS532Z on firmware 3.75 (362001000748+U-CS532Z(LT)V3.75.bin). Seems like the latest firmware is from 2023 ?
Buzzer_ON_OFF does not have any effect on the buzzer control. Does anybody know why?

**Comment by namezys** [comment]
> I'm currently using a CS532Z on firmware 3.75 (362001000748+U-CS532Z(LT)V3.75.bin). Seems like the latest firmware is from 2023 ? Buzzer_ON_OFF does not have any effect on the buzzer control. Does anybody know why?

You mast send `Buzzer_ON_OFF=1` TOGETHER with other parameters. It affects only current command instead of change state of a device.

As I can see, Gree had a bug in early versions

**Comment by xSzamanx** [comment]
> I will try with version 1.32 tomorrow, it is the version available on 'test' server, but it has same commProtVer as 1.31. There were reports of people being able to upgrade to 1.53 from 'test' server last year (for the same wifi board), this version has commProVer 3.0.0, both 1.21 and 1.23 have version set to 2.0.0 and 1.31 to 3.4M - but I'm not able to confirm if 1.53 silences the buzzer or not (yet)...

I'm usign Gree Fairy with 1.53 firmware and using Home Assistant Gree Climate integration. Silent commands works only every second command.

**Comment by namezys** [comment]
> I'm usign Gree Fairy with 1.53 firmware and using Home Assistant Gree Climate integration. Silent commands works only every second command.

wait for next firmware update. Looks like they fixed it

**Comment by simonepsp** [comment]
> > I'm currently using a CS532Z on firmware 3.75 (362001000748+U-CS532Z(LT)V3.75.bin). Seems like the latest firmware is from 2023 ? Buzzer_ON_OFF does not have any effect on the buzzer control. Does anybody know why?
> 
> You mast send `Buzzer_ON_OFF=1` TOGETHER with other parameters. It affects only current command instead of change state of a device.
> 
> As I can see, Gree had a bug in early versions

python3 gree.py -c 10.10.10.9 -i [redacted] -k "[redacted]" set SetTem=23 Buzzer_ON_OFF=1

still beeps :(

**Comment by xSzamanx** [comment]
> > I'm usign Gree Fairy with 1.53 firmware and using Home Assistant Gree Climate integration. Silent commands works only every second command.
> 
> wait for next firmware update. Looks like they fixed it

Which firmware do you mean?
As I remember for my ac's gree didn't deliver any update in their app. The only way for udpating my devices I should use modified app with test server firmwares.

**Comment by cerahmed** [comment]
> The only way for udpating my devices I should use modified app with test server firmwares.

Could you share which modified app are you using? And did the firmware succeed?
I'm having disconnection issues with my wifi unit, and dying to get the firmware to update.

**Comment by xSzamanx** [comment]
@cerahmed Unfortunetly I don't have that app now and I lost it. It was modified GREE app connecting to test server. I belive that there's somewhere an information how to compile such apk. This app shows and alows to update my two ac's.

**Comment by misomarcell** [comment]
Yep, `python gree.py -c <...> -i <...> -k <...> set Buzzer_ON_OFF=1 Pow=0 Lig=0` still beeps on Gree GWH12ACC 😠 

UPDATE: I've called the support and according to them, the buzzer simply cannot be disabled.

**Comment by misomarcell** [comment]
So here's the solution to the annoying beeper that cannot be turned off via software...

1. You need to remove the cover of the AC unit, including the door, body and internal electronics shield: https://i.imgur.com/Ii3fJab.jpeg
2. Locate the f*cker: https://i.imgur.com/8dMt76f.jpeg
3. Either de-solder it (100% chance to void warranty) or...

I've tried to use hot glue, but even a centimeter thick cover all around wasn't muting it noticeably, so I've used a toothpick that I plugged the buzzer's hole with, pressing against the buzzer's inner disc on one side and the plastic of the unit on the other, therefore keeping the buzzer's disc under pressure at all time. Then I fixed it in place with some more hot-glue: https://i.imgur.com/fbWO2AP.jpeg
( <100% chance to void warranty) 

The result is a ~90% reduction in the buzzer's volume, which is good enough for me.

**Comment by zhouska** [comment]
> [@cerahmed](https://github.com/cerahmed) Unfortunetly I don't have that app now and I lost it. It was modified GREE app connecting to test server. I belive that there's somewhere an information how to compile such apk. This app shows and alows to update my two ac's.

nope, that stopped working long time ago, that ship has sailed. The only way is to use something like mitmproxy and spoof the firmware URL. This would work, but I still don't know if the debug firmware (1.53 for my board) fixes the buzzer or not...

**Comment by zhouska** [comment]
> So here's the solution to the annoying beeper that cannot be turned off via software...
> 
>     1. You need to remove the cover of the AC unit, including the door, body and internal electronics shield: https://i.imgur.com/Ii3fJab.jpeg
> 
>     2. Locate the f*cker: https://i.imgur.com/8dMt76f.jpeg
> 
>     3. Either de-solder it (100% chance to void warranty) or...
> 
> 
> I've tried to use hot glue, but even a centimeter thick cover all around wasn't muting it noticeably, so I've used a toothpick that I plugged the buzzer's hole with, pressing against the buzzer's inner disc on one side and the plastic of the unit on the other, therefore keeping the buzzer's disc under pressure at all time. Then I fixed it in place with some more hot-glue: https://i.imgur.com/fbWO2AP.jpeg ( <100% chance to void warranty)
> 
> The result is a ~90% reduction in the buzzer's volume, which is good enough for me.

I read in some other thread you can in fact remove buzzer inner disc without breaking it, not sure where it was though...

**Comment by misomarcell** [comment]
> I read in some other thread you can in fact remove buzzer inner disc without breaking it, not sure where it was though...

The black plastic cover of it is one solid piece, cannot be opened from the top, maybe from the bottom, but for that you need to remove it from the board firs.

**Comment by Bruhtek** [comment]
> I just started using `gree.py` to control my Heiwa Essentiel ZEN+ (HMIS2-25P-V2) air conditioner. I was excited to find this topic, because the beeping is very annoying. Unfortunately my unit comes with firmware 1.23 and shows the same bug as mentioned above: Silencing the AC works only every other command.
> 
> Member ov1d1u was able to update his 1.23 to 1.31 three weeks ago and it fixed the issue with `Buzzer_ON_OFF`.
> 
> I checked the app, but there's no firmware update available for my unit.
> 
> I was intrigued by zhouska's comment to attempt to make the unit load a firmware from a different region. I tried this, but did not succeed. I can see the AC successfully downloading the 1.31 firmware, but after that nothing happens. The update progress in the app times out and everything returns to normal. The unit is not bricked, but it is not updated either.
> 
> For completeness, here's what I tried:
> 
>     * With `gree.py search --verbose` my unit returns `362001065279+U-WB05RT13V1.23.bin`.
> 
>     * I checked https://eugrih.gree.com/wifiModule/Lastversion/362001065279, but no firmware is available:
> 
> 
> ```
> {
>   "r":405,
>   "msg":"Firmware not found."
> }
> ```
> 
>     * However, https://kfgrih.gree.com/wifiModule/Lastversion/362001065279 returns 1.31!
> 
> 
> ```
> {
>   "CreateDate":"2025-02-25 07:44:09",
>   "commProtVer":"V3.4.M",
>   "desc":"ota upgrade",
>   "forcedUpgrade":0,
>   "r":200,
>   "url":"http://kf.grih.gree.com/wifiModule/image/14376/807236",
>   "ver":"1.31"
> }
> ```
> 
>     * Getting URL http://kf.grih.gree.com/wifiModule/image/14376/807236 downloads the firmware with following headers:
> 
> 
> ```
> HTTP/1.1 200 OK
> Date: Sat, 24 May 2025 17:10:53 GMT
> Content-Type: application/octet-stream
> Content-Length: 807236
> Connection: keep-alive
> Content-Disposition: attachment; filename="U-WB05RT13V1.31-17463628558052314052.bin"
> Content-Range: bytes 0-807235/807236
> FileMd5: 3f960fdfeaad1e344c14456a8caa2d59
> ```
> 
>     * The filename contains the very same `U-WB05RT13` as reported by my AC. So, I was starting to get confident the firmware would be compatible.
> 
>     * In mitmproxy I set `map_remote` = `|eugrih.gree.com/wifiModule|kfgrih.gree.com/wifiModule` to redirect the firmware update check from Europe to Asia (? not sure which region 'kf' is).
> 
>     * With the phone's traffic going through mitmproxy, I checked for firmware updates. Success! This time firmware 1.31 is shown as available for my device.
> 
>     * I started a packet capture of the air conditioner's IP address via ntopng on my pfSense router, just so that I could see what it does.
> 
>     * I started the firmware update in the app.
> 
>     * After a few minutes the progress got stuck at 99%. Another few minutes later the app returned to the previous screen as if nothing had happened.
> 
>     * After inspecting the packet capture, I could see that the AC downloaded the firmware successfully from http://kf.grih.gree.com/wifiModule/image/14376/807236 in about 90 seconds, but nothing happened after that. The unit did not brick, nor did it update. It just works as before. So, there must be something with the downloaded file or headers that the AC did not like. Or I must be missing something else. I don't know what, though...

I have the same AC model, but since I didn't want to bother with proxies, I went with a slightly simpler solution - I created a new account in Asia region, (same username,email,password - works fine), selected Taiwan as a region when prompted, and the update was available in the app 1.23 -> 1.31, same as you. Finished without a problem, and afterwards, I had to send the buzzer_on_off a few times, but once I did, it works perfectly now - it doesn't beep at all now, while previously it would beep every second time 

ETA: According to @mahi2003 's comment, this update makes it so that a dummy TLS server no longer works, and the AC needs internet access now, so be aware of that!

**Comment by krismarc** [comment]
> I have the same AC model, but since I didn't want to bother with proxies, I went with a slightly simpler solution - I created a new account in Asia region, (same username,email,password - works fine), selected Taiwan as a region when prompted, and the update was available in the app 1.23 -> 1.31, same as you. Finished without a problem, and afterwards, I had to send the buzzer_on_off a few times, but once I did, it works perfectly now - it doesn't beep at all now, while previously it would beep every second time 

Did you had to connect AC to wifi again?

**Comment by Bruhtek** [comment]
Yes, just reset the WiFi and add it in the app

**Comment by krismarc** [comment]
@Bruhtek, worked for me! Thank you very much for sharing the info!

![Image](https://github.com/user-attachments/assets/c175edb3-1f30-4eac-b45a-a8c8fe4e8229)

<img width="744" height="1334" alt="Image" src="https://github.com/user-attachments/assets/65985fbe-8cad-4697-9d2e-0fe93f6ee002" />

**Comment by Bruhtek** [comment]
Glad to hear that!

**Comment by mahi2003** [comment]
@Bruhtek, thank you! I was able to update my Heiwa air conditioner firmware from 1.23 to 1.31 with your instructions. I can confirm: No more beeping!

Unfortunately I seem to run into a new issue: My air conditioner is on a separate network without Internet access. Even with the older firmware I had to use a [gree dummy TLS server](https://codeberg.org/joserebelo/gree-dummy-tls-server) to fool the air conditioner in thinking it is connected to the cloud, otherwise it would disconnect or be unresponsive. After updating to 1.31 the workaround with the dummy server no longer seems to work. Exceptions are thrown in the dummy server log and the air conditioner is unresponsive. For now I have re-enabled Internet access. I will have to investigate.

**Comment by Bruhtek** [comment]
I've edited my comment to warn others of this, thank you!

**Comment by mahi2003** [comment]
I don't want to pollute this topic further, but after some investigation I can confirm that firmware 1.31 uses another protocol to talk to the cloud. This breaks existing dummy servers, like the one I am using. My findings so far: [Firmware 1.31: Gree Dummy TLS Server no longer working. New MQTT protocol?](https://codeberg.org/joserebelo/gree-dummy-tls-server/issues/3).

**Comment by ahmedragab202** [comment]
hello , i made "grok and claude" a script that searches and controls your in wifi netowrk AC units an all controls are in silent mode , thanks to your help and info from this issue thread 
t discovers devices, saves their details (IP, ID, key), and offers an interactive menu to manage:





Power, Mode (Auto/Cool/Dry/Fan/Heat), Temperature (16–30°C)



Fan Speed (Auto/Low–High), Quiet, Turbo



Swing, Sleep, Health, Anti-Freeze, Air, Blow modes



Display Light

Features:





Resolves fan conflicts (e.g., disables Quiet/Turbo for manual speeds).



Checks mode to ensure fan control (sets Cool if Auto).



Retries commands for reliability, logs actions to gree.log.



Termux-compatible with color-coded output. but will require an old 0.35 version of cryptography 

Usage: Save as ~/gree-remote/PythonCLI/gree.sh, run chmod +x gree.sh, then ./gree.sh. Select a device and use options 0–18.

[gree.txt](https://github.com/user-attachments/files/22065411/gree.txt)

its a sh file that should be placed in PythonCLI folder and run from inside

**Comment by ahmedragab202** [comment]
> hello , i made "grok and claude" a script that searches and controls your in wifi netowrk AC units an all controls are in silent mode , thanks to your help and info from this issue thread t discovers devices, saves their details (IP, ID, key), and offers an interactive menu to manage:
> 
> Power, Mode (Auto/Cool/Dry/Fan/Heat), Temperature (16–30°C)
> 
> Fan Speed (Auto/Low–High), Quiet, Turbo
> 
> Swing, Sleep, Health, Anti-Freeze, Air, Blow modes
> 
> Display Light
> 
> Features:
> 
> Resolves fan conflicts (e.g., disables Quiet/Turbo for manual speeds).
> 
> Checks mode to ensure fan control (sets Cool if Auto).
> 
> Retries commands for reliability, logs actions to gree.log.
> 
> Termux-compatible with color-coded output. but will require an old 0.35 version of cryptography
> 
> Usage: Save as ~/gree-remote/PythonCLI/gree.sh, run chmod +x gree.sh, then ./gree.sh. Select a device and use options 0–18.
> 
> [gree.txt](https://github.com/user-attachments/files/22065411/gree.txt)
> 
> its a sh file that should be placed in PythonCLI folder and run from inside



edit : it turns off turbo and quiet mode automatically as well when you want to change fan speed manually

**Comment by zhouska** [comment]
> > I just started using `gree.py` to control my Heiwa Essentiel ZEN+ (HMIS2-25P-V2) air conditioner. I was excited to find this topic, because the beeping is very annoying. Unfortunately my unit comes with firmware 1.23 and shows the same bug as mentioned above: Silencing the AC works only every other command.
> > Member ov1d1u was able to update his 1.23 to 1.31 three weeks ago and it fixed the issue with `Buzzer_ON_OFF`.
> > I checked the app, but there's no firmware update available for my unit.
> > I was intrigued by zhouska's comment to attempt to make the unit load a firmware from a different region. I tried this, but did not succeed. I can see the AC successfully downloading the 1.31 firmware, but after that nothing happens. The update progress in the app times out and everything returns to normal. The unit is not bricked, but it is not updated either.
> > For completeness, here's what I tried:
> > ```
> > * With `gree.py search --verbose` my unit returns `362001065279+U-WB05RT13V1.23.bin`.
> > 
> > * I checked https://eugrih.gree.com/wifiModule/Lastversion/362001065279, but no firmware is available:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > {
> >   "r":405,
> >   "msg":"Firmware not found."
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > * However, https://kfgrih.gree.com/wifiModule/Lastversion/362001065279 returns 1.31!
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > {
> >   "CreateDate":"2025-02-25 07:44:09",
> >   "commProtVer":"V3.4.M",
> >   "desc":"ota upgrade",
> >   "forcedUpgrade":0,
> >   "r":200,
> >   "url":"http://kf.grih.gree.com/wifiModule/image/14376/807236",
> >   "ver":"1.31"
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > * Getting URL http://kf.grih.gree.com/wifiModule/image/14376/807236 downloads the firmware with following headers:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > HTTP/1.1 200 OK
> > Date: Sat, 24 May 2025 17:10:53 GMT
> > Content-Type: application/octet-stream
> > Content-Length: 807236
> > Connection: keep-alive
> > Content-Disposition: attachment; filename="U-WB05RT13V1.31-17463628558052314052.bin"
> > Content-Range: bytes 0-807235/807236
> > FileMd5: 3f960fdfeaad1e344c14456a8caa2d59
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > * The filename contains the very same `U-WB05RT13` as reported by my AC. So, I was starting to get confident the firmware would be compatible.
> > 
> > * In mitmproxy I set `map_remote` = `|eugrih.gree.com/wifiModule|kfgrih.gree.com/wifiModule` to redirect the firmware update check from Europe to Asia (? not sure which region 'kf' is).
> > 
> > * With the phone's traffic going through mitmproxy, I checked for firmware updates. Success! This time firmware 1.31 is shown as available for my device.
> > 
> > * I started a packet capture of the air conditioner's IP address via ntopng on my pfSense router, just so that I could see what it does.
> > 
> > * I started the firmware update in the app.
> > 
> > * After a few minutes the progress got stuck at 99%. Another few minutes later the app returned to the previous screen as if nothing had happened.
> > 
> > * After inspecting the packet capture, I could see that the AC downloaded the firmware successfully from http://kf.grih.gree.com/wifiModule/image/14376/807236 in about 90 seconds, but nothing happened after that. The unit did not brick, nor did it update. It just works as before. So, there must be something with the downloaded file or headers that the AC did not like. Or I must be missing something else. I don't know what, though...
> > ```
> 
> I have the same AC model, but since I didn't want to bother with proxies, I went with a slightly simpler solution - I created a new account in Asia region, (same username,email,password - works fine), selected Taiwan as a region when prompted, and the update was available in the app 1.23 -> 1.31, same as you. Finished without a problem, and afterwards, I had to send the buzzer_on_off a few times, but once I did, it works perfectly now - it doesn't beep at all now, while previously it would beep every second time
> 
> ETA: According to [@mahi2003](https://github.com/mahi2003) 's comment, this update makes it so that a dummy TLS server no longer works, and the AC needs internet access now, so be aware of that!

Interesting find indeed. I presume everything else stays the same? No change to encryption key? I'm asking because cursory search using ``strings`` does NOT show the encryption keys (v1 nor v2) encoded in the firmware...

**Comment by ahmedragab202** [comment]
@zhouska 
If you don't want to update you can use my script in the comment before yours 
Place it in PythonCLI folder  and test it 
It gives buzzer on off =1 order with every command because even though all my AC units are 1.31 the buzzer order is not consistent and needs to be put in every command sent

**Comment by mahi2003** [comment]
@zhouska 
> Interesting find indeed. I presume everything else stays the same? No change to encryption key? I'm asking because cursory search using `strings` does NOT show the encryption keys (v1 nor v2) encoded in the firmware...

The device encryption key is generated each time the wifi module is paired with the cloud. Upgrading the firmware by itself will not change it, but in order to follow the instructions by Bruhtek you have to pair the wifi module with another account and that will change the device encryption key. At least until firmware 1.31 is published by Gree in your region.

@ahmedragab202
There's a bug in the older firmwares that makes the air conditioner still beep every other command, even when setting Buzzer_ON_OFF=1 with every command. Sending the same command twice will be silent for one and beep for the other. Firmware 1.31 fixes that.

**Comment by ahmedragab202** [comment]
@mahi2003 so it will beep in the second command even with silent command ?

**Comment by zhouska** [comment]
> [@zhouska](https://github.com/zhouska)
> 
> > Interesting find indeed. I presume everything else stays the same? No change to encryption key? I'm asking because cursory search using `strings` does NOT show the encryption keys (v1 nor v2) encoded in the firmware...
> 
> The device encryption key is generated each time the wifi module is paired with the cloud. Upgrading the firmware by itself will not change it, but in order to follow the instructions by Bruhtek you have to pair the wifi module with another account and that will change the device encryption key. At least until firmware 1.31 is published by Gree in your region.
> 
> [@ahmedragab202](https://github.com/ahmedragab202) There's a bug in the older firmwares that makes the air conditioner still beep every other command, even when setting Buzzer_ON_OFF=1 with every command. Sending the same command twice will be silent for one and beep for the other. Firmware 1.31 fixes that.

So in other words the v2 encryption method will still work from HA?

**Comment by zhouska** [comment]
> [@mahi2003](https://github.com/mahi2003) so it will beep in the second command even with silent command ?

I think he meant you need to send the command to disable buzzer with every request. It is not an one off toggle, but rather additional optional command, if you don§t include it, it will beep...

Having said that, the way it is worded in the script, is bit confusing:

```
# Function to send AC command with buzzer always on
send_ac_command() {
    local ip="$1"
    local id="$2"
    local key="$3"
    local params="$4"
    local full_params="$params Buzzer_ON_OFF=1"
    local cmd="python3 '$BASE_DIR/gree.py' -c '$ip' -i '$id' -k '$key' -e GCM set $full_params"
    run_gree_command "$cmd"
}
```

The ``Buzzer_ON_OFF=1`` in fact means it is disabled (it is very counterintuitive / confusing), so if anything, that method should have comment changed to something like 'always off'

**Comment by xSzamanx** [comment]
Unfortunetly not all Gree AC's are fixed with 1.31 firmware. I have console type Gree unit which beeps even with 1.31 firmware. It beeps with each command.

**Comment by ahmedragab202** [comment]
@zhouska I have 4 ACs on 1.31 and all of them have to get the command Buzzer_ON_OFF=1 in every command , the script adds it to every command , you can see the script does control everything , it also disables and enables turbo and quiet mode while making sure to disable the other mode if you want to choose manual speed 

The question is , does it work on 1.23 or anything less than 1.31 idk 
But yes I cannot persistently set buzzer to be off

**Comment by ahmedragab202** [comment]
@xSzamanx try my script , change to SH and put it in PythonCli 
And run and test it

**Comment by zhouska** [comment]
> [@zhouska](https://github.com/zhouska) I have 4 ACs on 1.31 and all of them have to get the command Buzzer_ON_OFF=1 in every command , the script adds it to every command , you can see the script does control everything , it also disables and enables turbo and quiet mode while making sure to disable the other mode if you want to choose manual speed
> 
> The question is , does it work on 1.23 or anything less than 1.31 idk But yes I cannot persistently set buzzer to be off

no, it does not, at least on the units with 362001065279 WiFi board (I have 1.23 and it suppresses every second beep only)

**Comment by mahi2003** [comment]
@zhouska
> So in other words the v2 encryption method will still work from HA?

Yes, firmware 1.31 works fine with Home Assisant. There were no changes to the local access.

It is not until you try to disconnect the air conditioner from the Internet that you will face issues...

@ahmedragab202
> so it will beep in the second command even with silent command ?

Yes indeed. That's the whole reason why several of us wanted to update to firmware 1.31. With the older firmware versions it is not possible to get rid of the beeping entirely.

But judging by xSzamanx' comment, 1.31 might not fix it for everyone either...

@xSzamanx 
Which wifi module do you have? Mine reports `362001065279+U-WB05RT13V1.31.bin` (was 1.23 before the firmware update). The physical module in the air conditioner has a label with `GRJWB04-J` and `WB05RT13V1.23`.

**Comment by ahmedragab202** [comment]
@zhouska that's sad

**Comment by ahmedragab202** [comment]
@mahi2003 I'm new to all this , I followed this thread and made the script just cause my grumpy rommemate kept turning turbo 16 celisus at night to make me want to leave and got angry when I modify and he hears the beep , now I can modify with no beeps  hehe joke on him

**Comment by zhouska** [comment]
> [@zhouska](https://github.com/zhouska)
> 
> > Interesting find indeed. I presume everything else stays the same? No change to encryption key? I'm asking because cursory search using `strings` does NOT show the encryption keys (v1 nor v2) encoded in the firmware...
> 
> The device encryption key is generated each time the wifi module is paired with the cloud. Upgrading the firmware by itself will not change it, but in order to follow the instructions by Bruhtek you have to pair the wifi module with another account and that will change the device encryption key. At least until firmware 1.31 is published by Gree in your region.
> 
> [@ahmedragab202](https://github.com/ahmedragab202) There's a bug in the older firmwares that makes the air conditioner still beep every other command, even when setting Buzzer_ON_OFF=1 with every command. Sending the same command twice will be silent for one and beep for the other. Firmware 1.31 fixes that.

You have same WiFi board as I do. Were you able to upgrade the firmware to 1.31 version as per @Bruhtek? If so, does it silence EVERY beep?

I don't care about being connected to the cloud that much, I use the EWPE app from time to time as well...

**Comment by ahmedragab202** [comment]
@zhouska is there a way I can control an AC that is not on the wifi network ?
Since I need to get it connected to wifi through the app 
can I control or connecto to it without having gree or tosot apps at all ?
Also if you change your app to tosot and region to Asia you get the update

**Comment by mahi2003** [comment]
@zhouska
> You have same WiFi board as I do. Were you able to upgrade the firmware to 1.31 version as per [@Bruhtek](https://github.com/Bruhtek)? If so, does it silence EVERY beep?

I confirm: No more beeps at all. You might have to toggle the beeper switch in Home Assistant a few times, but after that it is and remains totally silent.

I used the method described by Bruhtek to upgrade the firmware. After the upgrade, I performed another wifi reset and switched back to my old account in the European region, but that's optional.

@ahmedragab202
> is there a way I can control an AC that is not on the wifi network ? Since I need to get it connected to wifi through the app can I control or connecto to it without having gree or tosot apps at all ?

Before you can control the air conditioner via local access, it must be connected to wifi. For that you need the Gree app (or the rebranded app of your air conditioner). Once connected to wifi, you can control the air conditioner locally via the python script described in this github repo, or via Home Assistant, or perhaps other home automation systems. You then no longer need the Gree app. You can remove it from your phone if you wish.

**Comment by xSzamanx** [comment]
> [@xSzamanx](https://github.com/xSzamanx) try my script , change to SH and put it in PythonCli And run and test it

@ahmedragab202 Tried your script and still beeps with each command. 


> But judging by xSzamanx' comment, 1.31 might not fix it for everyone either...
> 
> [@xSzamanx](https://github.com/xSzamanx) Which wifi module do you have? Mine reports `362001065279+U-WB05RT13V1.31.bin` (was 1.23 before the firmware update). The physical module in the air conditioner has a label with `GRJWB04-J` and `WB05RT13V1.23`.

@mahi2003 Mine is also 362001065279+U-WB05RT13V1.31.bin

**Comment by zhouska** [comment]
> [@zhouska](https://github.com/zhouska)
> 
> > You have same WiFi board as I do. Were you able to upgrade the firmware to 1.31 version as per [@Bruhtek](https://github.com/Bruhtek)? If so, does it silence EVERY beep?
> 
> I confirm: No more beeps at all. You might have to toggle the beeper switch in Home Assistant a few times, but after that it is and remains totally silent.

Ok, I can also confirm the AC unit remains completely silent after upgrading to firmware 1.31 (this WiFi board: GRJWB04-J / 362001065279 / U-WB05RT13 ), the only downside is that it requires connection to the cloud which cannot be (for time being) defeated by https://codeberg.org/joserebelo/gree-dummy-tls-server

**Comment by sonic1200** [comment]
Hello, 
I've finally managed to have a working command silent on my gree Heatpump (vireo gen 3) with Home Assistant. I have a monkey patch to the normal Gree climate integration to supress the bip. (maybe not functionnal for every gree heat pump, but if you have the same model, I confirm that it works).
Here is the explanation that I wrote on my Gist : https://gist.github.com/sonic1200/ac1e41f68f7f73ded79eea8223b99857

**Comment by sonic1200** [comment]
I received a 1.44 firmware update today and since the hack used with Buzzer_ON_OFF=1  is no more working at all. It seems that they completely removed this undocumented command. Do you have an idea about other undocumented command which would still work with this recent firmware?
